lib/wanda/execution.ex

```diff
@@ -35,6 +29,14 @@ defmodule Wanda.Execution do
     end
   end
 
+  def get_checks_from_targets(targets) do
```

Comment:
The fact that this one needs to be public to be testable, and that it needs an integration test, it is a smell that maybe does not belong to this module. Perhaps we could move the function, without the `Catalog.get_check/1` call to the `Target` module and just test the pure function till uniq without the catalog side effects. Consider also adding `Catalog.get_checks/1` if needed.

Reply:
I agree that we can move the code to the Target module in order to segregate the responsability better, and we can test it better and faster :D